force-app/main/default/lwc/nksSurvey/nksSurvey.js

```diff
@@ -0,0 +1,214 @@
+import { LightningElement, track, wire } from 'lwc';
+import { ShowToastEvent } from 'lightning/platformShowToastEvent';
+import getSurvey from '@salesforce/apex/NKS_InternalSurveyController.getSurvey';
+import hasAnswered from '@salesforce/apex/NKS_InternalSurveyController.hasAnswered';
+import createAnsweredRecord from '@salesforce/apex/NKS_InternalSurveyController.createAnsweredRecord';
+import createResponseRecord from '@salesforce/apex/NKS_InternalSurveyController.createResponseRecord';
+export default class NksSurvey extends LightningElement {
+    surveyId;
+    backgroundColor;
+    title;
+    question;
+    isAnswered;
+    isRendered = false;
+    recordTypeId;
+    startDate;
+    endDate;
+
+    @track rating;
+    @track comment;
+    @track show = true;
+
+    @wire(getSurvey)
+    wiredSurvey({ data, error }) {
+        if (data) {
+            this.surveyId = data.Id;
+            this.backgroundColor = data.NKS_Background_Color__c;
+            this.title = data.NKS_Title__c;
+            this.question = data.NKS_Question__c;
+            this.startDate = data.NKS_Start_Date__c;
+            this.endDate = data.NKS_End_Date__c;
+        } else if (error) {
+            console.log('Problem getting survey: ' + error);
+        }
+    }
+
+    get background() {
+        return `background-color: ${this.backgroundColor}`;
+    }
+
+    get hide() {
+        return !this.show || this.isAnswered;
```

Comment:
Kan vi ikke bare bruke isAnswered hele tiden, istedenfor å ha ekstra this.show?

Reply:
show-variabelen er for å skjule komponenten etter bruker sender svar, isAnswered hentes etter neste refresh

Reply:
Stemmer, men kan vel bruke isAnswered for begge tilfeller og bare overskrive, istedenfor å introdusere en ny variabel? 😄

Reply:
Enig med Eirik her, hvis alt isAnswered er brukt til er å skjule komponenten så gir det ikke særlig mening her. Vi burde heller bare bruke show og sette den til true hvis vi ikke finner et answered record.